src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -168,86 +170,151 @@ static pthread_key_t g_threadKey;
 extern bool PalQueryProcessorTopology();
 bool InitializeFlushProcessWriteBuffers();
 
+extern "C" void RaiseFailFastException(PEXCEPTION_RECORD arg1, PCONTEXT arg2, UInt32 arg3)
+{
+    // Abort aborts the process and causes creation of a crash dump
+    abort();
+}
+
 void TimeSpecAdd(timespec* time, uint32_t milliseconds)
 {
-    time->tv_nsec += milliseconds * tccMilliSecondsToNanoSeconds;
-    if (time->tv_nsec > tccSecondsToNanoSeconds)
+    uint64_t nsec = time->tv_nsec + (uint64_t)milliseconds * tccMilliSecondsToNanoSeconds;
+    if (nsec >= tccSecondsToNanoSeconds)
     {
-        time->tv_sec += (time->tv_nsec - tccSecondsToNanoSeconds) / tccSecondsToNanoSeconds;
-        time->tv_nsec %= tccSecondsToNanoSeconds;
+        time->tv_sec += nsec / tccSecondsToNanoSeconds;
+        nsec %= tccSecondsToNanoSeconds;
     }
+
+    time->tv_nsec = nsec;
 }
 
+#ifdef __APPLE__
+// Convert nanoseconds to the timespec structure
+// Parameters:
+//  nanoseconds - time in nanoseconds to convert
+//  t           - the target timespec structure
+void NanosecondsToTimespec(uint64_t nanoseconds, timespec* t)
+{
+    t->tv_sec = nanoseconds / tccSecondsToNanoSeconds;
+    t->tv_nsec = nanoseconds % tccSecondsToNanoSeconds;
+}
+#endif // __APPLE__
+
+void ReleaseCondAttr(pthread_condattr_t* condAttr)
+{
+    int st = pthread_condattr_destroy(condAttr);
+    ASSERT_MSG(st == 0, "Failed to destroy pthread_condattr_t object");
+}
+
+class PthreadCondAttrHolder : public Wrapper<pthread_condattr_t*, DoNothing, ReleaseCondAttr, nullptr>
+{
+public:
+    PthreadCondAttrHolder(pthread_condattr_t* attrs)
+    : Wrapper<pthread_condattr_t*, DoNothing, ReleaseCondAttr, nullptr>(attrs)
+    {
+    }
+};
+
 class UnixEvent
 {
+    bool m_isValid;
```

Comment:
Nit - it would be better (less padding) for the bool to be together with other bools.